src/model/deep_links.rs

```diff
@@ -177,13 +179,26 @@ impl From<&Stream> for StreamDeepLinks {
                     URI_COMPONENT_ENCODE_SET
                 ),
             ),
+            external: match &stream.source {
+                StreamSource::Torrent { .. } => {
+                    if !streaming_server.settings.is_ready() { 
+                        stream_to_magnet_uri(stream)
+                    } else {
+                        "".to_owned()
+                    }
+                },
+                StreamSource::External { external_url } => {
+                    external_url.as_str().to_owned()
+                },
+                _ => "".to_owned()
```

Comment:
That match should be refactored after you implement the magent_url method.
`StreamSource::Torrent { .. }  if !streaming_server.settings.is_ready() => stream.magnet_url()`
`StreamSource::Url { url }  if !streaming_server.settings.is_ready() && url.scheme() == "magnet" => stream.magnet_url()`